security-analysis/security-analysis-api/src/main/java/com/powsybl/security/action/GeneratorAction.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.action;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An action to:
+ * - increase or decrease the targetV of a generator.
+ * - move to voltage control or remove from voltage control.
+ * - change targetQ of a generator that is not a voltage controller.
+ * - change targetV of a generator that is a voltage controller.
+ *
+ * @author Hadrien Godard <hadrien.godard@artelys.com>
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public class GeneratorAction extends AbstractAction {
```

Comment:
for the setter methods, I would avoid boxed types and use only primitive types in arguments.

If needed to unset some attributes I think API users would prefer to have unset methods instead of calling set with null.

Reply:
Actually, another design concern:
I would advise to not have setters at all in `Action` implementations:
from the point of view of the simulator, they should be immutable objects, the simulator is not supposed to change them.

If constructors (or static methods) are not enough and we want a fluent API maybe we should have a builder class ? (in which the comment above about the setter method applies :) )

Reply:
hi, not so sure about making these objects immutable. In simulators e.g. openloadflow indeed the simulator should not modify the actions.

but the same action objects defined by powsybl-core could be used to e.g. build a GUI, and the user might want to modify an action instead of having to delete the existing and create a new one.

Reply:
I understand the use case but in that case I think the best design will be to have a separate class in an application layer (where there might be some additional app-specific logic if needed), here the objects are really meant to be security analysis inputs: it's very important that the API does not allow the simulator to change them, or even other external components to change them during the execution of the simulator.
Otherwise, it will open doors for tricky bugs :)